### PR TITLE
Setup backend for cards

### LINKED ISSUE #417: Rework card and user ID
## Description
Rework card so that it is not reliant to Google token. It should instead have its own ID connected to the user ID.

## Acceptance criteria
- [x] The card is disconnected from Google and connected to its user via user ID instead.
- [x] Availability to change card for user.

## solution proposal
Create a separate table in the db for connecting card_id to user_id this can also contain metadata for the card.

### PR DESCRIPTION
**Pull request by Te4-Jacob-sandstrom**
## ⚙️ What issue did you implement or fix?
Fixes #417

## 📜 Summary
- Added cards with CRUD functionality to the backend
- Cards are connected to users via user_id
- Cards belong to Users

## 📝 Type of change

Please delete options that are not relevant.

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## 💩 How can this implementation be improved?
- This backend system needs to be used in the frontend so that book loans are connected to users via cards instead of directly. :shipit:

### CODE REVIEW

File: backend/app/controllers/api/v1/cards_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Api::V1::CardsController < ApplicationController
+  before_action :set_card, only: [:show, :edit, :update, :destroy]
+
+  # GET /cards
+  # GET /cards.json
+  def index
+    render json: Card.all
+    # return 'hhihi'
```
Comment (te4-adrian-almetun-smeds) [inline backend/app/controllers/api/v1/cards_controller.rb:8]: Umm ok. Maybe we can remove this.

File: backend/db/migrate/20201217141533_change_user_column_to_reference.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class ChangeUserColumnToReference < ActiveRecord::Migration[6.0]
+  def up
+    remove_column :cards, :user_uid
+    add_reference :cards, :user, type: :string, :class_name => 'User', :primary_key => 'uid'
+
+  end
+  
+  def down
+    
+  end
```
Comment (te4-adrian-almetun-smeds) [inline backend/db/migrate/20201217141533_change_user_column_to_reference.rb:10]:
```suggestion
```

File: backend/app/controllers/api/v1/cards_controller.rb (hunk 1 above)
Comment (itggot-linus-styren) [inline backend/app/controllers/api/v1/cards_controller.rb:8]: Can be removed
```suggestion
```

File: backend/db/seeds.rb [hunk 3]
```diff
@@ -140,6 +153,8 @@ def fixtures_review()
     fixtures_review
 when "subject"
     fixtures_subjects
+when "card"
```
Comment (itggot-linus-styren) [inline backend/db/seeds.rb:156]: Did you also run
```console
docker-compose run backend rails db:seed fixture=all
```

this is needed to update fixtures used in unit tests.

File: backend/test/fixtures/cards.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (itggot-linus-styren) [inline backend/test/fixtures/cards.yml:1]: See comment above on how to generate these fixtures automatically

File: backend/test/models/card_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class CardTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (te4-adrian-almetun-smeds) [inline backend/test/models/card_test.rb:6]:
```suggestion
```

**Review by itggot-linus-styren — CHANGES_REQUESTED** [review]
Looks good overall, just some minor comments.

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Can we replace all mentions of `expiration_data` with `expiration_date`?

This will be a nice feature when we can merge it.

File: backend/db/seeds.rb (hunk 3 above)
Comment (Te4-Jacob-sandstrom) [reply backend/db/seeds.rb:156]: I think I did that

**Comment by Te4-Jacob-sandstrom** [comment]
> Can we replace all mentions of `expiration_data` with `expiration_date`?
> 
> This will be a nice feature when we can merge it.

no

**Comment by te4-adrian-almetun-smeds** [comment]
> > Can we replace all mentions of `expiration_data` with `expiration_date`?
> > This will be a nice feature when we can merge it.
> 
> no

Why not?

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Will keep this comment until resolved: Can we replace all mentions of `expiration_data` with `expiration_date`?

**Review by te4-dennis-christensen — CHANGES_REQUESTED** [review]
what adrian said

**Review by te4-dennis-christensen — APPROVED** [review]
good job

**Review by te4-adrian-almetun-smeds — APPROVED** [review]
Now I can approve of this :+1:

**Review by te4-adrian-almetun-smeds — APPROVED** [review]
Now I can approve of this (again) 👍

**Review by itggot-linus-styren — APPROVED** [review]
LGTM

**Pull request merged**